### PR TITLE
vIRQ_{Dis,En}ableAll: Fix bug of never-enabled IRQs

### PR DESCRIPTION
**Pull request by AlessandroA**
When nesting multiple pairs of disable-all/enable-all function calls,
the last call to `vIRQ_EnableAll()` was not re-enabling all interrupts.

This was due to the fact that if multiple `vIRQ_DisableAll()` were called,
the wrong state `was_enabled = false` was set, even if the IRQ was
previously enabled.

Upon restoring the state, those IRQs were not re-enabled.

IRQs that were enabled during the critical section were correctly
enabled, because their state was unaffected.

This also improves performance because in the previous implementation
IRQs were always iterated over even if the counter was more than 0,
which is useless.

Fixes:

* bcf5fbe "Disable all IRQs: Added virtualised APIs".

This bug was captured by the new unit tests.

@meriac @Patater @niklas-arm

### CODE REVIEW

**Comment by Patater** [comment]
Before we move this feature into master, we should should apply this commit as a fixup to the original in the dev branch.

**Comment by Patater** [comment]
LGTM

File: core/system/src/unvic.c [hunk 1]
```diff
@@ -209,27 +209,29 @@ void unvic_irq_disable_all(void)
...
-    for (irqn = 0; irqn < NVIC_VECTORS; irqn++) {
-        if (g_unvic_vector[irqn].id == g_active_box && UNVIC_IS_IRQ_ENABLED(irqn)) {
-            /* Note: We only disable the IRQs if the counter is 0. */
-            if (!g_irq_disable_all_counter[g_active_box]) {
+    /* Only iterate over all IRQs if this is the first time that this function
+     * is called. The counter is always increased. */
+    if (g_irq_disable_all_counter[g_active_box] > 0) {
+        g_irq_disable_all_counter[g_active_box]++;
+    } else {
+        g_irq_disable_all_counter[g_active_box]++;
```
Comment (meriac) [inline core/system/src/unvic.c]: @AlessandroA fixme.

Comment (AlessandroA) [inline core/system/src/unvic.c]: Done.

**Pull request merged**